From 933d9fa5b6dfa8e810f292386f715a8b81632800 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Fri, 20 Sep 2019 19:41:10 +0300
Subject: [PATCH 671/741] enetc: Handle USXGMII protocol

commit 8580d65f83a20b94186ab05768afd2605819d7f9 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Adds USXGMII protocol which is now supported in Linux.  XGMII is kept for
compatibility although there is no plain XGMII support in ENETC.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index f1e03a79d5e3..2af56ec98420 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -529,7 +529,8 @@ static void enetc_configure_port_mac(struct enetc_hw *hw,
 		enetc_port_wr(hw, ENETC_PM0_IF_MODE, ENETC_PM0_IFM_RGAUTO);
 		enetc_port_wr(hw, ENETC_PM1_IF_MODE, ENETC_PM0_IFM_RGAUTO);
 	}
-	if (phy_mode == PHY_INTERFACE_MODE_XGMII) {
+	if (phy_mode == PHY_INTERFACE_MODE_XGMII ||
+	    phy_mode == PHY_INTERFACE_MODE_USXGMII) {
 		enetc_port_wr(hw, ENETC_PM0_IF_MODE, ENETC_PM0_IFM_XGMII);
 		enetc_port_wr(hw, ENETC_PM1_IF_MODE, ENETC_PM0_IFM_XGMII);
 	}
@@ -847,7 +848,8 @@ static int enetc_configure_serdes(struct enetc_ndev_priv *priv)
 	int err;
 
 	if (priv->if_mode != PHY_INTERFACE_MODE_SGMII &&
-	    priv->if_mode != PHY_INTERFACE_MODE_XGMII)
+	    priv->if_mode != PHY_INTERFACE_MODE_XGMII &&
+	    priv->if_mode != PHY_INTERFACE_MODE_USXGMII)
 		return 0;
 
 	err = enetc_imdio_init(pf);
@@ -857,7 +859,8 @@ static int enetc_configure_serdes(struct enetc_ndev_priv *priv)
 	if (priv->if_mode == PHY_INTERFACE_MODE_SGMII)
 		enetc_configure_sgmii(pf->imdio);
 
-	if (priv->if_mode == PHY_INTERFACE_MODE_XGMII)
+	if (priv->if_mode == PHY_INTERFACE_MODE_XGMII ||
+	    priv->if_mode == PHY_INTERFACE_MODE_USXGMII)
 		enetc_configure_sxgmii(pf->imdio);
 
 	return 0;
-- 
2.17.1

